tasks.py

```diff
@@ -1,28 +1,87 @@
 from invoke import task
+from invoke.exceptions import Failure
+from invoke.tasks import call
 
 
+def run_or_fail(c, cmd, fail_message="Error", fail_callback=None):
+    try:
+        cmd(c)
+    except Failure as ex:
+        print(fail_message)
+        if fail_callback:
+            fail_callback(c)
+        exit()
+
 @task()
+def tools_check(c, tools=[]):
+    missing_tools = []
+    for tool in tools:
+        if c.run(f"which {tool}", hide=True, warn=True).exited != 0:
+            missing_tools.append(f"{tool}")
+
+    if len(missing_tools) > 0:
+        print("These tools are required : {}".format(", ".join(missing_tools)))
+        print("Please install them")
+        exit()
+
+
+@task(pre=[call(tools_check, ["yarn"])])
 def setup_frontend(c):
     print("Running frontend setup...")
 
     with c.cd("frontend"):
         c.run("yarn")
 
 
-@task()
+@task(pre=[call(tools_check, ["docker"])])
+def setup_db(c):
+    print("Running DB setup...")
+
+    with c.cd("backend"):
+        run_or_fail(c, lambda x: x.run("ls .env", hide=True),
+                    fail_message="ERROR! Please copy backend/.env.sample "
+                                 "in backend/.env and set your .env vars")
+        with c.prefix(". .env"):
+            run_or_fail(c, lambda x: x.run("docker info", hide=True),
+                        fail_message="ERROR! Is Docker running?")
+            run_or_fail(c, lambda x: x.run("docker volume create $DOCKER_DB_VOLUME", hide=True),
+                        fail_message=f"ERROR! Docker volume '$DOCKER_DB_VOLUME' has not "
+                                     f"been created\n"
+                                     f"Here the results of command 'docker volume inspect $DOCKER_DB_VOLUME'",
+                        fail_callback=lambda x: x.run("docker volume inspect $DOCKER_DB_VOLUME"))
+            run_or_fail(c, lambda x: x.run("docker start $DOCKER_DB_CONTAINER ||"
```

Comment:
Maybe it makes sense to consider using docker-compose instead of the docker run command?

Reply:
So we can handle all the volumes etc stuff there, instead of running the various docker commands

Reply:
Yes @marcoacierno, i'm already working on it. I want to serve only db with docker-compose, is it ok?

Reply:
Hmm, it might definitely work as a start

Also I don't know if in the future we also want to move the backend on `docker-compose` and bind the local code when running with docker-compose.local.yml (for example). Or use docker-compose only for local deveploment as we have all the AWS stuff for production

Let me know if it's not clear what I'm thinking! :)

Reply:
Yes it's clear.
I agree with you